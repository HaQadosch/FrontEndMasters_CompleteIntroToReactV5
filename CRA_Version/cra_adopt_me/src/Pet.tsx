import React from 'react';

type PetProps = {
  name: string;
  animal: string;
  breed: string;
};

export const Pet: React.FC<PetProps> = ({ name, animal, breed }) => {
  return React.createElement('div', {}, [
    React.createElement('h1', {}, name),
    React.createElement('h2', {}, animal),
    React.createElement('h2', {}, breed),
  ]);
};
