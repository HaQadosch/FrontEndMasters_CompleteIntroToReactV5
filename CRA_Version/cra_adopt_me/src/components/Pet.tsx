import React from 'react';
import { Animal } from '@frontendmasters/pet';
import { Link } from '@reach/router';

export const Pet: React.FC<Animal> = ({
  id,
  name,
  type: animal,
  breeds: { primary: breed } = { primary: '' },
  photos: media = [],
  contact: { address } = { address: { city: '', state: '' } },
}) => {
  let hero = media.length ? media[0].small : 'http://placecorgi.com/300/300';
  return (
    <Link to={`/details/${id}`} className='pet'>
      <div className='image-container'>
        <img src={hero} alt={name} />
      </div>
      <div className='info'>
        <h1>{name}</h1>
        <h2>{`${animal} - ${breed} - ${address.city}, ${address.state}`}</h2>
      </div>
    </Link>
  );
};
