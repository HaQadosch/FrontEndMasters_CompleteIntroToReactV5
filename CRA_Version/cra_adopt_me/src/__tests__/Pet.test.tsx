import React from 'react';
import ReactDOM from 'react-dom';
import { render } from 'react-testing-library';
import { Pet } from '../Pet';

const testPet = {
  name: 'testName',
  animal: 'testAnimal',
  breed: 'testBreed',
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Pet {...testPet} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('Pet matches the snapshot', () => {
  const { debug, asFragment } = render(<Pet {...testPet} />);
  expect(asFragment()).toMatchSnapshot();
  debug(asFragment());
});
