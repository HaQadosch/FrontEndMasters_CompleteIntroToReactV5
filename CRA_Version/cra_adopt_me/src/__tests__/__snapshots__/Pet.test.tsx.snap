// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pet matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      testName
    </h1>
    <h2>
      testAnimal
    </h2>
    <h2>
      testBreed
    </h2>
  </div>
</DocumentFragment>
`;
